pandas/core/groupby/generic.py

```diff
@@ -1452,6 +1458,15 @@ def _transform_fast(self, result: DataFrame, func_nm: str) -> DataFrame:
         # for each col, reshape to to size of original frame
         # by take operation
         ids, _, ngroup = self.grouper.group_info
+
+        if any(
```

Comment:
I think this fix should be done in `group_info` instead of checking here

Reply:
@WillAyd this uses `result`, though, which wouldn't be available to `group_info`